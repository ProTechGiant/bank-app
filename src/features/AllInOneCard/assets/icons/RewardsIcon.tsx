import Svg, { Path } from "react-native-svg";

import type { IconProps } from "@/assets/icons";

export default function RewardsIcon({ width = 24, height = 24 }: IconProps) {
  return (
    <Svg xmlns="http://www.w3.org/2000/svg" width={width} height={height} viewBox="0 0 24 24" fill="none">
      <Path
        d="M1.22656 22.2705L15.2266 17.2705L6.22656 8.27047L1.22656 22.2705ZM11.5766 16.4505L4.52656 18.9705L7.04656 11.9205L11.5766 16.4505Z"
        fill="#FF371E"
      />
      <Path
        d="M13.7566 12.8005L19.3466 7.21047C19.8366 6.72047 20.6266 6.72047 21.1166 7.21047L21.7066 7.80047L22.7666 6.74047L22.1766 6.15047C21.1066 5.08047 19.3566 5.08047 18.2866 6.15047L12.6966 11.7405L13.7566 12.8005Z"
        fill="#FF371E"
      />
      <Path
        d="M9.28656 7.15047L8.69656 7.74047L9.75656 8.80047L10.3466 8.21047C11.4166 7.14047 11.4166 5.39047 10.3466 4.32047L9.75656 3.73047L8.69656 4.80047L9.28656 5.39047C9.76656 5.87047 9.76656 6.67047 9.28656 7.15047Z"
        fill="#FF371E"
      />
      <Path
        d="M16.2866 12.1505L14.6966 13.7405L15.7566 14.8005L17.3466 13.2105C17.8366 12.7205 18.6266 12.7205 19.1166 13.2105L20.7266 14.8205L21.7866 13.7605L20.1766 12.1505C19.0966 11.0805 17.3566 11.0805 16.2866 12.1505Z"
        fill="#FF371E"
      />
      <Path
        d="M14.2866 6.15047L10.6966 9.74047L11.7566 10.8005L15.3466 7.21047C16.4166 6.14047 16.4166 4.39047 15.3466 3.32047L13.7566 1.73047L12.6966 2.79047L14.2866 4.38047C14.7666 4.87047 14.7666 5.67047 14.2866 6.15047Z"
        fill="#FF371E"
      />
    </Svg>
  );
}
