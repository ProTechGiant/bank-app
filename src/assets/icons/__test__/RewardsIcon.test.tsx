import { create } from "react-test-renderer";

import { RewardsIcon } from "../RewardsIcon";

describe("icon/RewardsIcon", () => {
  it("renders a RewardsIcon", () => {
    const result = create(<RewardsIcon />);

    expect(result).toMatchSnapshot();
  });
});
