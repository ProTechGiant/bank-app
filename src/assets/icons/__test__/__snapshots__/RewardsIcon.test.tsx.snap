// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon/RewardsIcon renders a RewardsIcon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M10.6602 8.89L11.6002 12H14.4202L12.1502 13.62L13.0802 16.63L10.6602 14.79L8.24016 16.63L9.17016 13.62L6.90016 12H9.72016L10.6602 8.89ZM8.24016 10H0.660156L6.83016 14.41L4.49016 22L10.6602 17.31L16.8402 22L14.4902 14.41L20.6602 10H13.0802L10.6602 2L8.24016 10ZM21.0202 22L19.1602 15.99L23.3402 13H19.9002L16.8202 15.2L18.2802 19.92L21.0202 22ZM16.6602 8L14.8402 2L13.8002 5.45L14.5702 8H16.6602Z"
      fill={
        {
          "payload": 4294931730,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
